### PR TITLE
[LUA]Move PUP JA to jobUtils file

### PR DESCRIPTION
**Pull request by hooksta4**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?
Moves all PUP Job Abilities to job_utils lua.
The only change I made was 
https://github.com/LandSandBoat/server/blob/base/scripts/actions/abilities/cooldown.lua#L12

Where I added the check for if not automaton.

Looking at ways to do some trimming for repetitive code like checking for automaton. However, each of the onAbilityCheck for pup seems to have an additional check so it may need to stay how it is.  

image

- Updated based on the 3 CI check fails. 
- 
## Steps to test these changes
Unlock pup and use the job abilities.

### CODE REVIEW

File: scripts/actions/abilities/heady_artifice.lua [hunk 1]
```diff
@@ -8,13 +8,11 @@
...
 abilityObject.onAbilityCheck = function(player, target, ability)
-    ability:setRecast(math.max(0, ability:getRecast() - player:getMod(xi.mod.ONE_HOUR_RECAST) * 60))
-
-    return 0, 0
+    return xi.job_utils.puppetmaster.onAbilityCheckHeadyArtiface(player, target, ability)
 end
 
 abilityObject.onUseAbility = function(player, target, ability)
-    -- target:addStatusEffect(xi.effect.HEADY_ARTIFICE, 18, 1, 1) -- TODO: implement xi.effect.HEADY_ARTIFICE
+    return xi.job_utils.puppetmaster.onAbilityUseHeadyArtiface(player, target, ability)
```
Comment (xkoredev) [inline scripts/actions/abilities/heady_artifice.lua:15]: Is moving all these from smaller files to one central utils file a pattern that LSB wants to encourage? I have no strong opinions for or against, just curious about the motivation.

File: scripts/globals/job_utils/puppetmaster.lua [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+        pet:setHP(math.max(pet:getHP() * percent, 1))
+        pet:setMP(pet:getMP() * percent)
+    end
+end
+
+--Repair Function - TODO: verify that this function should be here and not on the items "Enhances Repair"
+-- https://www.bg-wiki.com/ffxi/Repair
+local function removeStatus(pet)
+    for _, effectId in ipairs(removableEffectIds) do
+        if pet:delStatusEffect(effectId) then
```
Comment (xkoredev) [inline scripts/globals/job_utils/puppetmaster.lua:122]: are you just moving code around?
Could rewrite this as `return pet:delStatusEffect(effectId)` (nit), but don't need to if the intent is just to move the code

File: scripts/globals/job_utils/puppetmaster.lua [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+            return true
+        end
+    end
+
+    if pet:eraseStatusEffect() ~= xi.effect.NONE then
```
Comment (xkoredev) [inline scripts/globals/job_utils/puppetmaster.lua:127]: same here, could rewrite as `return pet:eraseStatusEffect() ~= xi.effect.NONE`

File: scripts/actions/abilities/heady_artifice.lua (hunk 1 above)
Comment (hooksta4) [reply scripts/actions/abilities/heady_artifice.lua:15]: Yes - https://github.com/LandSandBoat/server/tree/base/scripts/globals/job_utils

File: scripts/globals/job_utils/puppetmaster.lua (hunk 2 above)
Comment (hooksta4) [reply scripts/globals/job_utils/puppetmaster.lua:122]: At this point, I am making no distinct changes to any of the code as far as calculations or process. However, there were a few places where I was able to trim some of the fat due to repeating checks or as an example, repair and maintenance. They have the same table of removable effects.

**Comment by xkoredev** [comment]
LGTM

File: scripts/globals/job_utils/puppetmaster.lua [hunk 4]
```diff
@@ -0,0 +1,410 @@
...
+    pet:wakeUp()
+
+    pet:delStatusEffect(xi.effect.REGEN)
+    pet:addStatusEffect(xi.effect.REGEN, regenAmount, 3, regenTime) -- 3 = tick, each 3 seconds.
+    player:removeAmmo()
+    return totalHealing
+end
+
+-- -- Maintenance Function
+-- local function removeStatus(target)
```
Comment (WinterSolstice8) [inline scripts/globals/job_utils/puppetmaster.lua]: Leftover code? if so remove this

**Comment by WinterSolstice8** [comment]
The CI failure isn't you, just need to rebase and you're good

**Review by WinterSolstice8 — APPROVED** [review]

**Comment by hooksta4** [comment]
> The CI failure isn't you, just need to rebase and you're good

Thanks - just was able to get some time to do it. Doing CI's now.

**Review by Xaver-DaRed — APPROVED** [review]

**Pull request merged**